### PR TITLE
48601: Packaging Workflow

### PR DESCRIPTION
**Pull request by jonquil2002**
SC: https://app.shortcut.com/replicated/story/48601/packaging-priorities-workflow

Previews:

Updated Distributing Workflow: https://deploy-preview-363--replicated-docs.netlify.app/vendor/distributing-workflow

Creating a Release: https://deploy-preview-363--replicated-docs.netlify.app/vendor/releases-creating-releases

Promoting Releases:  https://deploy-preview-363--replicated-docs.netlify.app/vendor/releases-promoting

Semantic Versioning (formerly Understanding Releases): https://deploy-preview-363--replicated-docs.netlify.app/vendor/releases-understanding

### CODE REVIEW

File: docs/vendor/distributing-workflow.md [hunk 1]
```diff
@@ -1,41 +1,126 @@
...
+        <th width="30%">Type</th>
+        <th width="70%">Description</th>
+      </tr>
+      <tr>
+        <td>Standard manifest files</td>
+        <td>We recommend using standard manifest YAML files unless you are already using Helm or Kubernetes Operators. Drag and drop the application files to the Replicated vendor portal or use the replicated CLI to import them.</td>
+      </tr>
+      <tr>
+        <td>Helm charts</td>
+        <td>If your application is already packaged using Helm charts, see <a href="helm-installing-replicated-helm">Installing with the App Manager.</a></td>
```
[the remark this answers is not in the document]
Comment (jonquil2002) [reply docs/vendor/distributing-workflow.md:40]: Some changes were made in a different PR, which automatically resolves the title issue/ I've changed it here to reflect the new title, which specifically says Helm.

File: docs/vendor/packaging-workflow.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+The following list is a suggested order, but you can configure these features in any order:
+
+**Note: This is a placeholder list. More info and content will be added.**
+
+1. Preflights
+1. Support bundles
+1. Database
+1. Port forwarding
+1. Ingress
+1. Blah, blah, blah
```
[the remark this answers is not in the document]
Comment (jonquil2002) [reply docs/vendor/packaging-workflow.md:77]: Ah. don't worry, this was a draft that we're not going to use. I presented two models to Paige and we decided to leave this one behind. I just hadn't deleted it yet, doing so now.

File: docs/vendor/releases-creating-releases.md [hunk 3]
```diff
@@ -37,3 +41,4 @@ We recommend that you bookmark the Replicated [vendor portal](https://vendor.rep
 ## Additional Resources
 
 [How to Distribute an Application](distributing-workflow)
+[How to Package an Application for Production](packaging-workflow)
```
[the remark this answers is not in the document]
Comment (jonquil2002) [reply docs/vendor/releases-creating-releases.md]: I deleted the latter.

File: docs/vendor/distributing-workflow.md [hunk 4]
```diff
@@ -1,41 +1,128 @@
...
+
+  :::note
+  Skip this step if your images are open-source or public, or if you are using Kubernetes Operators.
+  :::
+
+1. If you are providing your users with a cluster hosted on a VM, you must configure and test a Kubernetes installer. See <a href="packaging-embedded-kubernetes">Configure the Kubernetes Installer</a>.
+
+1. Create a license file in the vendor portal that contains entitlement information for your customer. You also need a license file to test your application in the admin console. See [Creating a Customer](releases-creating-customer).
+
+1. Promote the release and test it by installing the release in a development environment with the license file that you created. You can use the environment that you created during one of the recommended tutorials. For information about installation, see [Overview of Installing an Application](../enterprise/installing-overview).
```
Comment (ajp-io) [inline docs/vendor/distributing-workflow.md:60]: Is promoting considered part of packaging/distributing the release? Do we need to link to the docs on promoting a release? A similar question would be, in step 1, do we need to link to the section on creating a release? Or is it assumed those things are already known?

**Comment by ajp-io** [comment]
I don't know if it belongs in this PR or another, but I found that the first link on this page (about using the private registry) doesn't have the correct anchor anymore. Maybe the section name changed at some point. https://deploy-preview-363--replicated-docs.netlify.app/vendor/packaging-private-registry-security

File: docs/vendor/distributing-workflow.md [hunk 5]
```diff
@@ -1,41 +1,130 @@
...
-   * [Installing with an Existing Cluster](tutorial-installing-with-existing-cluster).
-   * [Installing without an Existing Cluster](tutorial-installing-without-existing-cluster).
-1. (Recommended) Learn about the application packaging process and begin to plan your deployment. See [Creating a Release](releases-creating-releases) and the topics in the _Packaging Your Application_ section.
-1. (Recommended) Send a questionnaire to your customers to gather information about their environments. See [Customer Application Deployment Questionnaire](planning-questionnaire).
-1. (Optional) Create a custom channel or edit the default channels in the vendor portal. See [Creating and Editing Channels](releases-creating-channels).
-1. Do the following to create, test, and edit a release:
-    1. Create a release in the vendor portal and write the Kubernetes manifest files to package your application. See [Creating a Release](releases-creating-releases).
+1. Push your images to the Replicated private registry if you plan to host your images that registry. See [Connecting to an Image Registry](packaging-private-images). Otherwise, proceed to the next step.
```
Comment (ajp-io) [inline docs/vendor/distributing-workflow.md:29]: Should this link point to this specific subsection instead? https://deploy-preview-363--replicated-docs.netlify.app/vendor/packaging-private-images#push-images-to-the-replicated-private-registry

File: docs/vendor/distributing-workflow.md [hunk 6]
```diff
@@ -1,41 +1,130 @@
...
+        <th width="30%">Type</th>
+        <th width="70%">Description</th>
+      </tr>
+      <tr>
+        <td>Standard manifest files</td>
+        <td>We recommend using standard manifest YAML files unless you are already using Helm or Kubernetes Operators. <br></br><br></br>To import using the Replicated vendor portal, see <a href="releases-creating-releases">Create a Release</a>. To import using the replicated CLI, see <a href="tutorial-installing-with-cli">Managing Releases with the CLI</a>.</td>
+      </tr>
+      <tr>
+        <td>Helm charts</td>
+        <td>If your application is already packaged using Helm charts, see <a href="helm-release">Adding Helm Charts to a Release</a>.</td>
```
Comment (ajp-io) [inline docs/vendor/distributing-workflow.md:44]: Link doesn't work

**Review by ajp-io — CHANGES_REQUESTED** [review]
A comment and one link that doesn't work

**Comment by jonquil2002** [comment]
@ajp-io The broken link to Helm is bc the topic file name was changed in another PR and my branch isn't pulling that in from `main` for some reason. The link will work when this PR is merged. If not, I can open another PR to fix it. If you can approve this, I will merge and test that right away.

**Review by ajp-io — APPROVED** [review]

**Pull request merged**